### PR TITLE
Travis CI: Add flake8 to find syntax errors and undefined names

### PR DESCRIPTION
**Pull request by cclauss**
Add [flake8](http://flake8.pycqa.org) tests to find Python syntax errors and undefined names.

__E901,E999,F821,F822,F823__ are the "_showstopper_" flake8 issues that can halt the runtime with a SyntaxError, NameError, etc. Most other flake8 issues are merely "style violations" -- useful for readability but they do not effect runtime safety.
* F821: undefined name `name`
* F822: undefined name `name` in `__all__`
* F823: local variable name referenced before assignment
* E901: SyntaxError or IndentationError
* E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree

### CODE REVIEW

**Comment by AetherUnbound** [comment]
This package does not currently support python 3.6 on a number of levels. Please remove it from the travis config as a matrix option.

**Comment by cclauss** [comment]
The Python 3.6 tests are in __allow_failures__ mode to aid in the porting process without halting the builds.  474 days until the end of life of Python 2.

**Comment by AetherUnbound** [comment]
Additionally, it looks like you've added `matrix` in twice to the config which is overriding our previous `matrix` config. Feel free to keep the 3.6 builds, but please add them to the `matrix` that already exists.
And believe me, we are well aware of the deadline :grinning:

**Comment by cclauss** [comment]
Sames tests run against all CellProfiler repos:
* <sub>https://travis-ci.org/CellProfiler/CellProfiler/builds/428840598?utm_source=github_status&utm_medium=notification</sub>

**Comment by AetherUnbound** [comment]
:tada: Wahoo! This looks good to me now @cclauss, anything else needed?

**Comment by cclauss** [comment]
Nice work.  Well done.

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +29,15 @@ notifications:
...
     on_failure: always
     on_success: change
-python: 2.7.12
+python:
+  - 2.7
+  - 3.6    
+before_script:
+  - pip install flake8
+  # stop the build if there are Python syntax errors or undefined names
+  - flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
```
[the remark this answers is not in the document]
Comment (cclauss) [reply .travis.yml]: Freezing the flake8 options in a separate .flake8 file would remove our flexibility to make the second flake8 run just below.

**Pull request merged**